app/services/container_deployment_service.rb

```diff
@@ -0,0 +1,60 @@
+class ContainerDeploymentService
+
+  def all_data
+    {
+      :provision              => possible_provision_providers,
+      :providers              => possible_providers_and_vms,
+      :cloud_init_template_id => cloud_init_template_id,
+    }.compact
+  end
+
+  def possible_provision_providers
+    providers = ExtManagementSystem.includes(:vm_or_template).select do |m|
+      m.instance_of?(ManageIQ::Providers::Amazon::CloudManager) || m.instance_of?(ManageIQ::Providers::Redhat::InfraManager)
+    end
+    providers.map do |provider|
+      {:provider => provider, :templates => templates(provider.miq_templates)}
+    end
+  end
+
+  def templates(templates)
+    templates.map do |template|
+      {
+        :cpu       => template.cpu_total_cores,
+        :memo      => template.mem_cpu,
```

Comment:
just a small nit but can you maybe change memo to mem or memory? to maintain consistency